### PR TITLE
common: Fix access to sa_family_t in FreeBSD

### PR DESCRIPTION
**Pull request by wjwithagen**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/msg/msg_types.h [hunk 1]
```diff
@@ -265,9 +268,7 @@ struct entity_addr_t {
     type = o.type;
     nonce = o.nonce;
     memcpy(&u, &o.in_addr, sizeof(u));
-#if !defined(__FreeBSD__)
```
Comment (tchaikov) [inline src/msg/msg_types.h]: @wjwithagen this change partially reverts 07b9bc03105b0e7b5ce3c90af2d686475f8f2657 by @stass . on FreeBSD, i guess `sa_family` is sorta `uint8_t`, per https://github.com/freebsd/freebsd/blob/master/sys/sys/_types.h#L61 . so i think, `ntohs(a_byte)` is buggy here. 

could you elaborate the rationale behind your fix in the commit message? as the problem it is addressing is not obvious to me.

Comment (wjwithagen) [reply src/msg/msg_types.h]: @tchaikov 
Not filing the family field generates trouble in other places where the socket needs to be opened, and sa_family is not set. I think it solved my problem with `rbd-ggate`, but not with `smoke.sh`.
This issue is there if I try and run tests with only msgr1 as authentication.
Will look into this further.

Comment (wjwithagen) [reply src/msg/msg_types.h]: @tchaikov. That is exactly what i was going to look into. And see if it does not “swapout” the byte of the family with a zero from the msb.

Comment (wjwithagen) [reply src/msg/msg_types.h]: @tchaikov 
As far as I can tell is `typedef __uint8_t  __sa_family_t` s on FreeBSD (/usr/include/sys/_types.h:61)
And on Linux it equals `typedef unsigned short __kernel_sa_family_t`  (include/uapi/linux/socket.h:11)

So I need the LSB of of something that is only 1 byte ;-) 
closing this, and looking for other reasons that do not give me an AF_<value> in 
`src/msg/async/net_handler.cc:36: int NetHandler::create_socket(int domain, bool reuse_addr)`

**Pull request closed without merging**